source/js/modules/letters-animation.js

```diff
@@ -0,0 +1,86 @@
+export default () => {
+
+    class AccentTypographyBuild {
+        constructor(
+            elementSelector,
+            timer,
+            classForActivate,
+            property
+        ) {
+            this._TIME_SPACE = 100;
+
+            this._elementSelector = elementSelector;
+            this._timer = timer;
+            this._classForActivate = classForActivate;
+            this._property = property;
+            this._element = document.querySelector(this._elementSelector);
+            this._timeOffset = 100;
+
+            this.prePareText();
+        }
+
+        createElement(letter, index) {
+            const span = document.createElement(`span`);
+            span.textContent = letter;
+            span.style.transition = `${this._property} ${this._timer}ms ease ${this._timeOffset}ms`;
+            // this._timeOffset += 20;
+            // if ((index % 2) == 0) {
+            //     this._timeOffset += 100;
+            //   } else {
+            //     this._timeOffset -= 100;
+            //   }
+            if ((index % 5) == 0) {
+                this._timeOffset = 300;
+            } else if ((index % 3) == 0) {
+                this._timeOffset = 200;
+            } else if ((index % 2) == 0) {
+                this._timeOffset = 250;
+            } else {
+                this._timeOffset = 150;
+            }
+            return span;
+        }
+
+        prePareText() {
+            if (!this._element) {
+                return;
+            }
+            const text = this._element.textContent.trim().split(` `).filter((latter) => latter !== '');
```

Comment:
Можно чуть упростить:
```suggestion
            const text = this._element.textContent.trim().split(` `).filter(Boolean);
```